src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -34,7 +34,7 @@
  * @author Kévin Dunglas <dunglas@gmail.com>
  * @author Samuel ROZE <samuel.roze@gmail.com>
  */
-final class PaginationExtension implements ContextAwareQueryResultCollectionExtensionInterface
+final class PaginationExtension implements ContextAwareQueryResultCollectionExtensionInterface, ContextAwareQueryCollectionExtensionInterface
```

Comment:
Thing is ContextAwareQueryResultCollectionExtensionInterface extends QueryCollectionExtensionInterface and changing the interface would go against the BC promise see https://symfony.com/doc/current/contributing/code/bc.html#changing-interfaces (at least I think). We can't really do this for now :/